src/packagedcode/go_sum.py

```diff
@@ -0,0 +1,133 @@
+rsc.io/quote v1.5.2 h1:w5fcysjrx7yqtD/aO+QwRjYZOKnaM9Uh2b40tElTs3...
+rsc.io/quote v1.5.2/go.mod h1:LzX7hefJvL54yjefDEDHNONDjII0t9xZLPX...
+rsc.io/sampler v1.3.0 h1:7uVkIFmeBqHfdjD+gZwtXXI+RODJ2Wc4O7MPEh/Q...
+rsc.io/sampler v1.3.0/go.mod h1:T1hPZKmBbMNahiBKFy5HrXp6adAjACjK9...
+
+"""
+
+"""
+go.sum file contains certain things in 2 formats.
+
+For example:
+github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+
+
+>>> ob = GoSum()
+>>> p = ob.parse_dep_type2('github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=')
+>>> assert p.group('namespace') == ('github.com/BurntSushi')
+>>> assert p.group('name') == ('toml')
+>>> assert p.group('version') == ('v0.3.1')
+>>> assert p.group('checksum') == ('WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=')
+
+>>> p = ob.parse_dep_type1('github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=')
+>>> assert p.group('namespace') == ('github.com/BurntSushi')
+>>> assert p.group('name') == ('toml')
+>>> assert p.group('version') == ('v0.3.1')
+>>> assert p.group('checksum') == ('xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=')
+"""
+
+
+TRACE = False
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+@attr.s()
+class GoSum(object):
+    # Regex expressions to parse different types of dependency
+    # dep_type1 example: github.com/BurntSushi/toml v0.3.1 h1:WXkYY....
+    parse_dep_type1 = re.compile(
+        r'(?P<namespace>(.*))'
+        r'\/'
+        r'(?P<name>[^\s]*)'
+        r'(\s)*'
+        r'(?P<version>(.*))'
+        r'/go.mod(\s)*h1:'
+        r'(?P<checksum>(.*))'
+    ).match
+
+    # dep_type2 example: github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCN....
+    parse_dep_type2 = re.compile(
+        r'(?P<namespace>(.*))'
+        r'/'
+        r'(?P<name>[^\s]*)'
+        r'(\s)*'
+        r'(?P<version>(.*))'
+        r'(\s)*h1:'
+        r'(?P<checksum>(.*))'
+    ).match
+
+    @classmethod
+    def parse_gosum(cls, location):
+        """
+        Return a list containing all the go.sum dependency data.
+        """
+        with io.open(location, encoding='utf-8', closefd=True) as data:
+            lines = data.readlines()
+
+        gosum_data = []
+
+        for line in lines:
+            parsed_dep = cls.parse_dep_type1(line)
+            if not parsed_dep:
+                parsed_dep = cls.parse_dep_type2(line)
+
+            dep_data = [parsed_dep.group('namespace').strip(),
+                    parsed_dep.group('name').strip(),
+                    parsed_dep.group('version').strip()]
+
+            gosum_data.append(dep_data)
+
+        gosum_data.sort()
+        gosum_data = list(gosum_data for gosum_data, _ in itertools.groupby(gosum_data))
```

Comment:
What is this doing?

Reply:
@pombredanne It was handling dupes. This removes the duplicates form the list of lists

Reply:
OK, may you need to find a way to do this in a less clever, but easier to read and understand way? Or at the minimum add some explanatory comment?